Emma good branch

Emma3DSS opened the pull request:
---
name: Pull Request
about: Suggest merging your contribution

---

**Describe what your contribution is**
A clear and concise description of what you have made.

**Show what you've made**
If your contribution is in any way visual, art or otherwise, include a screenshot

**List any issue that it relates to**
If your contribution is result of a bug or suggestion in the issue, remember to cross reference them here.

**Changelog**
Below is the format for the changelog. Please use it. Spacing is important.

### Changelog

[changelog]

* [Added] New feature
* [Fixed] A bug
* [Changed] Something
* [Removed] Something broken

[/changelog]

Dyhr commented:
Post processing is a package now, use that rather than adding it from the asset store

Dyhr commented:
You added materials in the Assets/Models/Blender folder. Can you keep that folder clean of anything not blender files?

Dyhr commented:
And let me know if I'm wrong. Github might not be showing it right

Dyhr commented:
Assets/Models/Models.fbx sounds like you just exported everything. Can you export it as separate .fbx files?

Dyhr commented:
Music is nice and all, but please keep pull requests to a small scope in the future. It's fine though

Dyhr commented:
Finally, be slightly more specific in your changelog

    [Added] WIP Map
    [Added] New Posters
    [Added] New Poster Textures // This is the same as above
    [Added] New Tiles & Tile Textures/Colors
    [Added] New Models // What models
    [Added] New Materials/Particles // Be slightly more specific. Don't group particles with materials, they are not that related
    [Added] Particle System // Where? Show it off or tell us what it is
    [Added] Post-Processing // This is in the package manager now
    [Added] Main SS13 Theme // Could be cut. Keep it simple
    [Added] Top Down Directional Lighting // Wasn't this already there?
    [Fixed] Wall & Tile Sizes // Doesn't make sense since walls didn't exist before this

Emma3DSS commented:
> Post processing is a package now, use that rather than adding it from the asset store

Yeah I could not find the package in manager last night so I just installed from github I will delete and install from package manager

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+{
+
+    public float damage = 10f;
+    public float recoil = -10f;
```
I would prefer to have this be a positive number, for usability in the inspector

Emma3DSS replied:
Resolved this

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+
+    public Rigidbody gun;
+    public Camera GunCamera;
+    public AudioSource mySound;
```
These fields can be private

Emma3DSS replied:
Having them as public allows the script to be interchangeable far more easily then having them private. Different Public Variables for Different guns

Dyhr replied:
Use [SerializeField] instead

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+    public float recoil = -10f;
+    public float range = 100f;
+
+    public Rigidbody gun;
```
You do not need a rigid body. Just use a transform, and it should specifically be the muzzle, not the gun itself

Emma3DSS replied:
?Resolved?

Dyhr replied:
Is it?

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+    public float range = 100f;
+
+    public Rigidbody gun;
+    public Camera GunCamera;
```
What is the camera for?

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+
+    public Rigidbody gun;
+    public Camera GunCamera;
+    public AudioSource mySound;
```
Don't name anything "myX". "audioSource" will do

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+    public ParticleSystem muzzleFlash;
+
+
+    // Update is called once per frame
```
Read this https://github.com/F-O-T-S/Client/blob/master/StyleGuides/C_SHARP.md

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+
+
+    // Update is called once per frame
+    void Update()
```
Add explicit private

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+        {
+
+            Shoot();
+
```
I would remove some of the excess white space, but it's not critical

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+
+    }
+
+    void Shoot()
```
This should be public, most likely

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+        RaycastHit hit;
+
+        muzzleFlash.Play();
+        gun.AddForce(recoil, 0, 0, ForceMode.Impulse);
```
The recoil should be optional. Check if there is a rigidbody and recoil if there is. Less assumptions is better

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+        gun.AddForce(recoil, 0, 0, ForceMode.Impulse);
+        mySound.Play();
+
+        if (Physics.Raycast(GunCamera.transform.position, GunCamera.transform.forward, out hit, range))
```
A mask would be great, so we don't raycast stuff we don't need to

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+        {
+
+            
+            Debug.Log(hit.transform.name);
```
This should do something rather than just logging

Emma3DSS replied:
I like to know what the gun is hitting.

Dyhr replied:
And I'd like something to happen in the game

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+
+        if (hit.rigidbody)
+        {
+            Vector3 direction = hit.rigidbody.transform.position - transform.position;
```
direction doesn't seem to be used

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+        if (hit.rigidbody)
+        {
+            Vector3 direction = hit.rigidbody.transform.position - transform.position;
+            hit.rigidbody.AddForceAtPosition(GunCamera.transform.forward * damage, hit.point);
```
Force should be separate from damage. Optionally it can be forcePerDamage

Dyhr commented on Assets/Scripts/Gun.cs:
```diff
@@ -0,0 +1,53 @@
+    void Update()
+    {
+
+        if (Input.GetKeyDown(KeyCode.E))
```
Put this in a separate script clearly named something with test. That makes it easier to clean up later.

Emma3DSS replied:
yeah mb it was an accident

Dyhr commented on ProjectSettings/InputManager.asset:
```diff
@@ -293,3 +293,19 @@ InputManager:
     axis: 0
     joyNum: 0
+  - serializedVersion: 3
+    m_Name: UseKey
```
Do we really need to add anything to the input? Can't we just hard code a key for testing?

Emma3DSS replied:
That was an accident sorry

Dyhr commented:
If the gun was a separate pull request, the map could probably have been in by now

Dyhr commented:
I cleaned up the room for you, but didn't put everything back.

Dyhr commented:
We're kinda waiting on proper walls

Dyhr commented:
The MuzzleFlash folder needs a new home

Dyhr commented:
Particles do not belong in materials

Emma3DSS commented:
> I cleaned up the room for you, but didn't put everything back.



> The MuzzleFlash folder needs a new home

Fixed

> Particles do not belong in materials

No need to change only materials in there

Emma3DSS commented on Assets/Scripts/Gun.cs:
```diff
@@ -1,17 +1,23 @@
-    public AudioSource mySound;
+    public Transform GunBarrel;
     public ParticleSystem muzzleFlash;
+    public AudioSource mySound;
```
Resolved

Emma3DSS commented on Assets/Scripts/Gun.cs:
```diff
@@ -1,17 +1,23 @@
     public Rigidbody gun;
-    public Camera GunCamera;
-    public AudioSource mySound;
+    public Transform GunBarrel;
```
Resolved

Emma3DSS commented on Assets/Scripts/Gun.cs:
```diff
@@ -32,10 +38,10 @@ void Shoot()
 
         muzzleFlash.Play();
-        gun.AddForce(recoil, 0, 0, ForceMode.Impulse);
+        gun.AddForce(0, recoil, 0, ForceMode.Impulse);
```
yea I was just making the recoil go a different way

Dyhr commented:
Oh, I think you missed some comments
